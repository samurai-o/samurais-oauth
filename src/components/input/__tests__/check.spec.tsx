import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";
import Input from "..";

describe("checkbox 组件测试", () => {
	test("checkbox 快照", () => {
		const dom = render(<Input.Checkbox>猜测</Input.Checkbox>);
		expect(dom.asFragment()).toMatchSnapshot();
	});
});