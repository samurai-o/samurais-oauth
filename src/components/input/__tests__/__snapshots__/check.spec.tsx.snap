// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox 组件测试 checkbox 快照 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-left: 10px;
  color: #fff;
}

.c1 {
  cursor: pointer;
  position: relative;
  width: 18px;
  height: 18px;
  font-size: 12px;
  visibility: hidden;
  overflow: visible;
  color: inherit;
  font-family: inherit;
  line-height: inherit;
  margin: 0;
}

.c1::after {
  position: absolute;
  top: 0px;
  border-radius: 10px;
  border: 2px solid #ddd;
  background-color: transparent;
  color: #000;
  width: 18px;
  height: 18px;
  box-sizing: border-box;
  display: inline-block;
  visibility: visible;
  text-align: center;
  content: ' ';
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
}

.c1:checked::after {
  font-family: system-ui;
  content: '✓';
  font-size: 12px;
  line-height: 1.2;
  text-align: center;
  color: #000;
  font-weight: bold;
  border: 2px solid #000;
  background-color: transparent;
}

<div
    class="c0"
  >
    <input
      class="c1"
      type="checkbox"
    />
    <label
      class="c2"
    >
      猜测
    </label>
  </div>
</DocumentFragment>
`;
